File: Java/Maths/SimpleNumber.java
/*
What is a simple number?
A simple number is a positive integer greater than 1 that has no positive divisors other than 1 and itself.

Explanation of Approach:
- We first check if the given number is less than or equal to 1. If yes, then it is not simple.
- Otherwise, we iterate from 2 to the square root of the given number and check if it is divisible by any number in that range. If yes, then it is not simple.
- If the number is not divisible by any number in the above range, then it is simple.

Time Complexity: The time complexity of this code is O(sqrt(n)), where n is the given number.

Space Complexity: The space complexity is O(1) as we are not using any extra space that depends on the input.

Sample Input: 17

Sample Output: 17 is a simple number
*/

public class SimpleNumber {

    public static boolean isSimple(int number) {
        // Numbers less than or equal to 1 are not simple
        if (number <= 1) {
            return false;
        }
        
        // Check if number is divisible by any number from 2 to square root of the number
        for (int i = 2; i <= Math.sqrt(number); i++) {
            if (number % i == 0) {
                return false;
            }
        }
        
        return true;
    }

    public static void main(String[] args) {
        int input = 17; // Sample input
        
        if (isSimple(input)) {
            System.out.println(input + " is a simple number");
        } else {
            System.out.println(input + " is not a simple number");
        }
    }
}

File: Python/Maths/SimpleNumber.py
'''
What is a simple number?
A simple number is a positive integer greater than 1 that has no positive divisors other than 1 and itself.

Approach:
1. The function `is_simple_number` takes an integer `n` as input.
2. First, we check if the number is less than 2. If it is, we return `False` as numbers less than 2 are not prime numbers.
3. Next, we loop from 2 to the square root of the number (inclusive) using the `range` function.
4. For each number in the loop, we check if the number is divisible by `i`. If it is, we return `False` as it is not a prime number.
5. If the number is not divisible by any number, we return `True` as it is a prime number.

Sample Input:
number = 13

Sample Output:
13 is a simple number

Time Complexity: The time complexity of this code is O(sqrt(n)). This is because we iterate from 2 to the square root of the number, which reduces the number of iterations needed compared to checking all numbers up to n.

Space Complexity: The space complexity of this code is O(1), as it only uses a constant amount of memory to store the input number and the loop variable.
'''
  
#Determining whether a number is simple or not in Python:

def is_simple_number(n):
    # Check if the number is less than 2, which is not a prime number
    if n < 2:
        return False
    
    # Check if the number is divisible by any number from 2 to square root of the number
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    
    # If the number is not divisible by any number, it is a prime number
    return True


# Sample Input
number = 13

# Check if the number is simple or not
is_simple = is_simple_number(number)

# Print the result
if is_simple:
    print(number, "is a simple number")
else:
    print(number, "is not a simple number")

